functions/Set-DbaDbSequence.ps1

```diff
@@ -0,0 +1,159 @@
+    param (
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Database,
+        [Parameter(Mandatory)]
+        [string]$Name,
+        [string]$Schema = 'dbo',
+        [long]$RestartWith,
+        [long]$IncrementBy,
+        [long]$MinValue,
+        [long]$MaxValue,
+        [switch]$Cycle,
+        [int32]$CacheSize,
+        [parameter(ValueFromPipeline)]
+        [Microsoft.SqlServer.Management.Smo.Database[]]$InputObject,
+        [switch]$EnableException
+    )
+    process {
+
+        if ((Test-Bound -ParameterName SqlInstance) -and (Test-Bound -Not -ParameterName Database)) {
+            Stop-Function -Message "Database is required when SqlInstance is specified"
+            return
+        }
+
+        foreach ($instance in $SqlInstance) {
+            $InputObject += Get-DbaDatabase -SqlInstance $instance -SqlCredential $SqlCredential -Database $Database
+        }
+
+        foreach ($db in $InputObject) {
+
+            if ($Pscmdlet.ShouldProcess($db.Parent.Name, "Modifying the sequence $Name in the $Schema schema in the database $($db.Name) on $($db.Parent.Name)")) {
+                try {
+                    $sequence = $db | Get-DbaDbSequence -Schema $Schema -Name $Name
+
+                    if ($null -eq $sequence) {
+                        Stop-Function -Message "Unable to find sequence $Name in the $Schema schema in the database $($db.Name) on $($db.Parent.Name)" -Continue
+                    }
+
+                    if (Test-Bound IncrementBy) {
+                        $sequence.IncrementValue = $IncrementBy
+                    }
+
+                    $sequence.IsCycleEnabled = $Cycle.IsPresent
+
+                    if (Test-Bound RestartWith) {
+                        $sequence.StartValue = $RestartWith # SMO does the restart logic when this value is changed and then Alter() is called (i.e. CurrentValue is also updated)
+                    }
+
+                    if (Test-Bound MinValue) {
+                        $sequence.MinValue = $MinValue
+                    }
+
+                    if (Test-Bound MaxValue) {
+                        $sequence.MaxValue = $MaxValue
+                    }
+
+                    if (Test-Bound CacheSize) {
+                        if ($CacheSize -eq 0) {
+                            $sequence.SequenceCacheType = [Microsoft.SqlServer.Management.Smo.SequenceCacheType]::NoCache
+                        } else {
+                            $sequence.SequenceCacheType = [Microsoft.SqlServer.Management.Smo.SequenceCacheType]::CacheWithSize
+                            $sequence.CacheSize = $CacheSize
+                        }
+                    } else {
+                        $sequence.SequenceCacheType = [Microsoft.SqlServer.Management.Smo.SequenceCacheType]::DefaultCache
+                    }
+
+                    $sequence.Alter()
+                    $sequence.Refresh()
+                    $sequence
```

Comment:
This should get the sequence from the instance again and not depend on the cached object/connection.

Reply:
ok, will do